packages/sign-client/src/controllers/engine.ts

```diff
@@ -1343,6 +1447,7 @@ export class Engine extends IEngine {
       throw new Error(message);
     }
     const { id, reason } = params;
+    this.checkRecentlyDeleted(id);
```

Comment:
I don't think this is covered in tests atm, only string-based topic `set/get` for the `recentlyDeleted` map. 

Are we sure this will behave as assumed here when using the proposal id (i.e. `number`) as a key?